### PR TITLE
fix cmake install target to have machine independent target files

### PR DESCRIPTION
**Pull request by dkuegler**
This is an update to a previous patch for svn, now adapted for my git repository. It was partially implemented in "* re #1010 Installing header files to include directory, adding xml files to share directory" ff (72303f4ae043bb74da9b8151df64a638b512cf5c ff.).

c.f. https://app.assembla.com/spaces/plus/messages/5692383

I should also be pointed out, that the commit introduces install path variables to th cmake to make install path handling easier (CMakeLists.txt) as well as installs some Testing/Utilities Targets.

Please do comment, and I would make some changes if needed. This is to get a "common" version, so I can maintain my version of PlusLib easier.

### CODE REVIEW

**Comment by adamrankin** [comment]
Please run <repoFolder>/Utilities/SetupForDevelopment.sh. This will configure the rules necessary for accepting pull requests.

File: src/PlusCalibration/CMakeLists.txt [hunk 1]
```diff
@@ -155,21 +161,22 @@ IF(PLUSBUILD_BUILD_PlusLib_TOOLS)
...
 # Install
 #
-INSTALL(TARGETS vtkPlusCalibration
-  RUNTIME DESTINATION bin CONFIGURATIONS Release COMPONENT RuntimeLibraries
-  LIBRARY DESTINATION lib CONFIGURATIONS Release COMPONENT RuntimeLibraries
-  ARCHIVE DESTINATION lib CONFIGURATIONS Release COMPONENT Development
+INSTALL(TARGETS vtkPlusCalibration EXPORT PlusLib
+  RUNTIME DESTINATION "${PLUSLIB_BINARY_INSTALL}" CONFIGURATIONS Release COMPONENT RuntimeLibraries
+  LIBRARY DESTINATION "${PLUSLIB_LIBRARY_INSTALL}" CONFIGURATIONS Release COMPONENT RuntimeLibraries
+  ARCHIVE DESTINATION "${PLUSLIB_LIBRARY_INSTALL}" CONFIGURATIONS Release COMPONENT Development
```
Comment (adamrankin) [inline src/PlusCalibration/CMakeLists.txt]: Could you add PLUSLIB_ARCHIVE_INSTALL, in case we want to change the location in the future.

**Comment by dkuegler** [comment]
Unfortunately, I saw the SetupForDevelopment.sh after the commit. Since I do not know, if you have "imported" the commit yet, so I am reluctant to change the commit message.

**Comment by adamrankin** [comment]
Hey David,

I'm going to do a thorough read through to make sure I understand all the changes. Give me a few hours and we'll make this happen.

**Comment by dkuegler** [comment]
Sure.

I have fixed some errors as time went by… Do tell if anything pops up…

Cheers
David


Von: Adam Rankin [mailto:notifications@github.com]
Gesendet: Freitag, 3. Februar 2017 17:32
An: PlusToolkit/PlusLib <PlusLib@noreply.github.com>
Cc: Kügler, David <david.kuegler@gris.informatik.tu-darmstadt.de>; Author <author@noreply.github.com>
Betreff: Re: [PlusToolkit/PlusLib] fix cmake install target to have machine independent target files (#1)


Hey David,

I'm going to do a thorough read through to make sure I understand all the changes. Give me a few hours and we'll make this happen.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -22,7 +22,7 @@ ENDIF()
 OPTION(PLUS_USE_OpenIGTLink "Use the OpenIGTLink protocol" OFF)
 SET(PLUS_OPENIGTLINK_VERSION)
 IF(PLUS_USE_OpenIGTLink)
-  FIND_PACKAGE(OpenIGTLinkIO REQUIRED NO_MODULE)
+  FIND_PACKAGE(OpenIGTLink REQUIRED NO_MODULE)
```
Comment (dzenanz) [inline src/CMakeLists.txt]: No need to change this

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -31,6 +31,8 @@ IF(PLUS_USE_OpenIGTLink)
   SET(PLUS_OPENIGTLINK_VERSION_MAJOR "${OpenIGTLink_VERSION_MAJOR}")
   SET(PLUS_OPENIGTLINK_VERSION_MINOR "${OpenIGTLink_VERSION_MINOR}")
   SET(PLUS_OPENIGTLINK_VERSION_PATCH "${OpenIGTLink_VERSION_PATCH}")
+  
+  FIND_PACKAGE(OpenIGTLinkIO REQUIRED NO_MODULE)
```
Comment (dzenanz) [inline src/CMakeLists.txt]: No need to change this

**Comment by adamrankin** [comment]
It is worth it. I had to take some time to understand the OpenIGTLinkIO install (which is similar to this one).
I've just gotten swamped with research work, and haven't had time to get back to it.

**Comment by dkuegler** [comment]
I am happy to help as well. I will update the pull request as soon as I get to it...

**Comment by dkuegler** [comment]
I updated my pull request to the current PlusLib master. It build without errors for me, but I have not been able to do intensive testing. The install file seems to work for my installation project as well.

File: src/PlusDataCollection/CMakeLists.txt [hunk 4]
```diff
@@ -1,1624 +1,1678 @@
...
+OPTION (PLUS_USE_BKPROFOCUS_CAMERALINK "Provide support for BK ProFocus ultrasound scanners through CameraLink interface" OFF)
+OPTION (PLUS_USE_ICCAPTURING_VIDEO "Provide support for the IC framegrabber device" OFF)
+OPTION (PLUS_USE_VFW_VIDEO "Provide support for the Video-for-Windows video digitizer" OFF)
+OPTION (PLUS_USE_EPIPHAN "Provide support for the Epiphan" OFF)
+OPTION (PLUS_USE_CAPISTRANO_VIDEO "Provide support Capistrano Labs USB ultrasound probes" OFF)
+OPTION (PLUS_USE_INTERSON_VIDEO "Provide support Interson USB ultrasound probes" OFF)
+OPTION (PLUS_USE_INTERSONSDKCXX_VIDEO "Provide support Interson SDK 1.X with C++ Wrapper USB ultrasound probes" OFF)
+OPTION (PLUS_USE_TELEMED_VIDEO "Provide support for the Telemed ultrasound probes" OFF)
+OPTION (PLUS_USE_THORLABS_VIDEO "Provide support for the ThorLabs compact spectrometers" OFF)
+OP
```
Comment (adamrankin) [inline src/PlusDataCollection/CMakeLists.txt]: Should this not be COMPONENT Development?

**Pull request merged**